src/test/java/unit/BqDatasetLifetime.java

```diff
@@ -0,0 +1,227 @@
+    assertEquals(partitionLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.PARTITION));
+    assertEquals(tableLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+
+    // Now set both to zero in a single call and make sure both get cleared.
+
+    updateDatasetWithDefaultLifetimes(
+        name, "--default-partition-lifetime=0", "--default-table-lifetime=0");
+
+    defaultLifetimeMap = getDefaultLifetimes(bqDataset);
+    assertEquals(0, defaultLifetimeMap.get(DefaultLifetimeEntity.PARTITION));
+    assertEquals(0, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+
+    // Test that flags can be passed along with generic controlled resource update flags by renaming
+    // the dataset (then renaming it back) along with a default lifetime update.
+
+    final String renamed = name + "Renamed";
+
+    updateDatasetWithDefaultLifetimes(
+        name,
+        "--new-name=" + renamed,
+        "--default-partition-lifetime=" + partitionLifetime.toString());
+    defaultLifetimeMap = getDefaultLifetimes(bqDataset);
+    assertEquals(partitionLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.PARTITION));
+    assertEquals(0, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+
+    updateDatasetWithDefaultLifetimes(
+        renamed, "--new-name=" + name, "--default-table-lifetime=" + tableLifetime.toString());
+    defaultLifetimeMap = getDefaultLifetimes(bqDataset);
+    assertEquals(partitionLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.PARTITION));
+    assertEquals(tableLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+  }
+
+  /**
+   * Helper method to get the name of the dataset that should be created by the test method (derived
+   * from the passed TestInfo instance).
+   */
+  private String getDatasetName(TestInfo testInfo) {
+    return testInfo.getTestMethod().orElseThrow().getName();
+  }
+
+  /** Creates a BQ Dataset with the given name and any passed lifetime arguments. */
+  private UFBqDataset createDatasetWithLDefaultLifetimes(
+      String name, String... defaultLifetimeArguments) throws JsonProcessingException {
+
+    String datasetId = randomDatasetId();
+
+    List<String> arguments =
+        new ArrayList<>(
+            Arrays.asList(
+                "resource", "create", "bq-dataset", "--name=" + name, "--dataset-id=" + datasetId));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+
+    return TestCommand.runAndParseCommandExpectSuccess(
+        UFBqDataset.class, arguments.toArray(new String[0]));
+  }
+
+  /** Updates a dataset with a given name using the passed arguments. */
+  private void updateDatasetWithDefaultLifetimes(String name, String... defaultLifetimeArguments) {
+    List<String> arguments =
+        new ArrayList<>(Arrays.asList("resource", "update", "bq-dataset", "--name=" + name));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+    TestCommand.runCommandExpectSuccess(arguments.toArray(new String[0]));
```

Comment:
`resource update` should also return a JSON object. Can you add a check for the fields returned there in one of your tests as well?